server/app/sessionStatus.js

```diff
@@ -98,7 +100,7 @@ module.exports.getSubmissionHistory = function (context, body, res) {
           arrNoNulls[i - shift] = current;
         }
       }
-      to_send[cohort] = {history: arrNoNulls.slice(0, arrNoNulls.length - shift), count: count};
+      to_send[cohort] = {history: arrNoNulls.slice(0, arrNoNulls.length - shift), count: count, data_1: data[1], data_0:data[0] };
```

Comment:
I am not 100% sure if I am correct, but it seems like the userkey is being sent next to the date of submission back to the analyst. if thats the case this is a problem: the analyst will be able to tell who submitted when. Furthermore, it is important that the analyst can never link the userkey (which they know which of the input companies it belongs to) to the jiff party_id which is randomly selected by the server, and only the server knows to which userkey it belongs to.

This here could be an attack vector, even if the information is not displayed to the analyst, if the analyst can use a browser inspector to view incoming data etc.

I am not sure what data_0 and data_1 are.

It is ok to use the userkey server side to find the most recent submission etc. But it is important that it is removed from the array before it is sent to the analyst. The easiest way to check this is to log the response analyst side and make sure it is anonymous.

Reply:
I think this was a feature request the BWWC made, and they accepted the privacy tradeoff for the ability to check who hasn't submitted data. 

Do you problems other than knowing which parties already submitted?

Reply:
Can confirm, this was requested by BWWC. We could probably make this configurable while we merge the pacesetters version too.

Reply:
OK, if it is a feature request it is what it is. I do not think there are other problems. I looked and the associated jiff party id is not revealed or associated with the revealed userkey.